src/main/java/com/cappella/csv/CsvClient.java

```diff
@@ -0,0 +1,214 @@
+import com.cappella.model.SubTask;
+import com.cappella.model.TaskData;
+
+public class CsvClient {
+
+    public static final String TYPE = "text/csv";
+
+    public static final String GRANT_HEADER_TASK_NAME = "Opportunity name";
+    public static final String GRANT_HEADER_DUE_DATE = "Funder Full proposal deadline";
+    public static final String GRANT_HEADER_SECTION = "Status";
+    public static final String GRANT_HEADER_SUBTASK_NAME = "Next task description";
+    public static final String GRANT_HEADER_SUBTASK_DUE_DATE = "Next task deadline";
+
+    final Logger LOGGER = LoggerFactory.getLogger(getClass());
+
+    /**
+     * @param file
+     * @return boolean
+     */
+    public boolean hasCSVFormat(MultipartFile file) {
+        if (!TYPE.equals(file.getContentType())) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * CSV file must have a first record/row of headers.
+     * The header "Opportunity name" (GRANT_HEADER_TASK_NAME) must exist.
+     * Will return a null List if either of these conditions are not met or
+     * there are errors parsing the headers.
+     * List my be empty if the header is present but no valid rows are parsed.
+     * Any errors encountered will be contained in the Problems.
+     * 
+     * @param is
+     * @param problems
+     * @return List<TaskData>
+     */
+    public List<TaskData> parseGrantCsvToTasks(InputStream is, Problems problems) {
+        List<TaskData> tasks = null;
+        try {
+            BufferedReader fileReader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+            CSVParser csvParser = new CSVParser(fileReader, CSVFormat.DEFAULT.withFirstRecordAsHeader().withTrim());
+            Iterable<CSVRecord> csvRecords = csvParser.getRecords();
+            // ensure there is at least the header used for the task name
+            if (!csvParser.getHeaderNames().contains(GRANT_HEADER_TASK_NAME)) {
+                // log error that the task name header does not exist in the csv file
+                problems.addError(Problems.ERROR_MISSING_GRANT_TASK_NAME_HEADER, null);
+            } else {
+                tasks = new ArrayList<TaskData>();
+                for (CSVRecord csvRecord : csvRecords) {
+                    TaskData task = parseTask(csvRecord, problems);
+                    if (task != null) {
+                        tasks.add(task);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            // handle all exceptions gracefully log error
+            problems.addError(Problems.ERROR_PARSING_CSV_FILE, e.toString());
+        }
+        return tasks;
+    }
+
+    /**
+     * Will return a null TaskData if the CSVRecord does not have a task.
+     * Any errors encountered will be contained in the Problems.
+     * 
+     * @param csvRecord
+     * @param problems
+     * @return
+     */
+    TaskData parseTask(CSVRecord csvRecord, Problems problems) {
```

Comment:
I have a habit of leaving them package private so I could make tests for each one of them if so desired.  One of my TODO items is to find a way to mock CSVRecord so I could test these directly.

Reply:
Back into religious territory here. Lots of people will do this. I was always taught that your tests should only go through your public interface, otherwise your unit tests get coupled to your implementation and make the class hard to change.